python/lsst/summit/utils/tmaUtils.py

```diff
@@ -0,0 +1,1346 @@
+        )
+
+
+class TMAState(enum.IntEnum):
+    """Overall state of the TMA.
+
+    States are defined as follows:
+
+    UNINITIALIZED
+        We have not yet got data for all relevant components, so the overall
+        state is undefined.
+    STOPPED
+        All components are on, and none are moving.
+    TRACKING
+        We are tracking the sky.
+    SLEWING
+        One or more components are moving, and one or more are not tracking the
+        sky. This should probably be called MOVING, as it includes: slewing,
+        MOVING_POINT_TO_POINT, and JOGGING.
+    FAULT
+        All (if engineeringMode) or any (if not engineeringMode) components are
+        in fault.
+    OFF
+        All components are off.
+    """
+    UNINITIALIZED = -1
+    STOPPED = 0
+    TRACKING = 1
+    SLEWING = 2
+    FAULT = 3
+    OFF = 4
+
+    def __repr__(self):
+        return f"TMAState.{self.name}"
+
+
+class AxisMotionState(enum.IntEnum):
+    """Motion state of azimuth elevation and camera cable wrap.
+
+    Note: this is copied over from
+    https://github.com/lsst-ts/ts_idl/blob/develop/python/lsst/ts/idl/enums/MTMount.py  # noqa: W505
+    to save having to depend on T&S code directly. These enums are extremely
+    static, so this is a reasonable thing to do, and much easier than setting
+    up a dependency on ts_idl.
+    """
+
+    STOPPING = 0
+    STOPPED = 1
+    MOVING_POINT_TO_POINT = 2
+    JOGGING = 3
+    TRACKING = 4
+    TRACKING_PAUSED = 5
+
+
+class PowerState(enum.IntEnum):
+    """Power state of a system or motion controller.
+
+    Also used for motion controller state.
+
+    Note that only a few systems (and no motion controllers)
+    use TURNING_ON and TURNING_OFF. The oil supply system is one.
+
+    Note: this is copied over from
+    https://github.com/lsst-ts/ts_idl/blob/develop/python/lsst/ts/idl/enums/MTMount.py  # noqa: W505
+    to save having to depend on T&S code directly. These enums are extremely
+    static, so this is a reasonable thing to do, and much easier than setting
+    up a dependency on ts_idl.
+    """
+
+    OFF = 0
+    ON = 1
+    FAULT = 2
+    TURNING_ON = 3
+    TURNING_OFF = 4
+    UNKNOWN = 15
+
+
+def getAxisAndType(rowFor):
+    """Get the axis the data relates to, and the type of data it contains.
+
+    Parameters
+    ----------
+    rowFor : `str`
+        The column in the dataframe denoting what this row is for, e.g.
+            "elevationMotionState" or "azimuthInPosition", etc.
+
+    Returns
+    -------
+    axis : `str`
+        The axis the row is for, e.g. "azimuth", "elevation".
+    rowType : `str`
+        The type of the row, e.g. "MotionState", "SystemState", "InPosition".
+    """
+    regex = r'(azimuth|elevation)(InPosition|MotionState|SystemState)'
+    matches = re.search(regex, rowFor)
+    axis = matches.group(1)
+    rowType = matches.group(2)
+
+    assert rowFor.endswith(f"{axis}{rowType}")
```

Comment:
I've added the `$` at the end, but I'm going to keep the assert because I'm awful at regex and that will help me be sure I'm getting what I want/help me understand my own code in future.